src/org.hdfgroup.hdfview/hdf/view/TableView/DataProviderFactory.java

```diff
@@ -1672,6 +1680,79 @@ public Object getDataValue(Object obj, int index)
 
             return theValue;
         }
+
+        /**
+         * update the data value of a compound type.
+         *
+         * @param columnIndex the column
+         * @param rowIndex    the row
+         * @param newValue    the new data value object
+         */
+        @Override
+        public void setDataValue(int columnIndex, int rowIndex, Object newValue)
+        {
+            Object theValue = newValue;
+            if (isFLT16)
+                theValue = Short.toString(Float.floatToFloat16(Float.parseFloat((String)newValue)));
+
+            super.setDataValue(columnIndex, rowIndex, theValue);
+        }
+
+        /**
+         * When a CompoundDataProvider wants to pass a List of data down to a nested CompoundDataProvider, or
+         * when a top-level container DataProvider (such as an ArrayDataProvider) wants to hand data down to a
+         * base CompoundDataProvider, we need to pass down a List of data and the new value, plus a field and
+         * row index. This method is for facilitating this behavior.
+         *
+         * In general, all "container" DataProviders that have a "container" base DataProvider should call
+         * down into their base DataProvider(s) using this{}, method, in order to ensure that buried
+         * CompoundDataProviders get handled correctly. When their base DataProvider is not a "container"
+         * type, the method setDataValue(index, Object, Object) should be used instead.
+         *
+         * For atomic type DataProviders, we treat this method as directly calling into setDataValue(index,
+         * Object, Object) using the passed rowIndex. However, this method should, in general, not be called
+         * by atomic type DataProviders.
+         *
+         * @param columnIndex the column
+         * @param rowIndex    the row
+         * @param bufObject   the data object
+         * @param newValue    the new data object
+         */
+        public void setDataValue(int columnIndex, int rowIndex, Object bufObject, Object newValue)
```

Comment:
I'm not sure that we actually need to implement this `setDataValue` variant here, and it may actually be problematic to do so, based on my old comment in the Javadoc above.

Reply:
Maybe, but container types will need the same method if I'm correct. Will need a test file.